### ISSUE TITLE
Possible solutions for Update/EOL

### ISSUE THREAD

**Issue by nachfuellbar** [telegramdesktop/tdesktop#5160]
Hello,

i don't know if this is the right platform for proposing different possbilities.
In Issue 4926 @john-preston admitted he doesn't know how to handle future EOL of "supported" distributions.
I thought quite some time about this but only 2 real solutions came up to me.

1) getting Telegram Desktop in upstream repositories of distributions (will take some efforts but is partly already done; for fedora at least rpmfusion is providing a package which doesn't has the slow start problematic - but thats another issue)

2) dividing updates in beta and stable channels - people only get autoupdate in stable if a certain amount of users is using it without issues (either by active acknowledgement or bei passive - if after certain timeframe at least X users are using it on distribution Y and don't open issues)

Maybe there are other/better solutions for this but i couldn't come up with something better.
First solution would pretty much exclude windows.
Second solutions is used in many software development projects. Often without real user interaction for choosing stable/beta (i experience this every time google tests a new android app - i get the new app/feature, many others not)

I hope this will help you ;)

**Comment by john-preston** [comment]
@nachfuellbar Thanks for your thoughts.

There is no problem with Windows and macOS. Those systems have linear versioning. I can just update the app to some version that will know that it's not updating any further if it was run on XP / Vista / OS X 10.6 / whatever. Or that it will update but in some other way, so we still are able to push some critical patches.

Linux is a different story with all its zoo of kernel versions / desktop environments / window managers. You don't even know how to segregate them.

I've upgraded my Linux development environment only once so far: from Ubuntu 12.04 to Ubuntu 14.04. When I did that the app stopped launching on Ubuntu 12.04 because of a new version of libc that was used on Ubuntu 14.04. It took some work to make a version built on 14.04 to still run on 12.04 (so I could autoupdate it) and I'm not sure I'll be able to do that for a next upgrade as well.

**Comment by nachfuellbar** [comment]
If Windows/Mac is no problem, then coming from my first proposition there would be something similiar.
There are several projects which provide own repositories for the mainstream distributions - they build their software for each destination in an own chroot or container (e.g. docker) 
But this would increase your overhead for providing updates quite a bit (at least if you don't use an CI/CD).
I don't even know how difficult it would be to provide a source tree which builds on every mainstream distribution as i am not a programmer (just it administrator/engineer)

**Comment by ilya-fedin** [comment]
@Aokromes this issue can be closed since preston added deprecating by glibc version
